src/unit_tests/wazuh_modules/wmparser/test_wm_parser.c

```diff
@@ -0,0 +1,168 @@
+        ":azure_wodle: - CRITICAL - This is a Critical\n"
+        };
+
+    will_return(__wrap_isDebug, 1);
+
+    expect_string(__wrap__mtinfo, tag, WM_AZURE_LOGTAG);
+    expect_string(__wrap__mtinfo, formatted_msg, "Received and acknowledged 0 messages");
+
+    expect_string(__wrap__mterror, tag, WM_AZURE_LOGTAG);
+    expect_string(__wrap__mterror, formatted_msg, "This is an Error\n:azure_wodlexx: - WARNING - This is a Warning");
+
+    expect_string(__wrap__mterror, tag, WM_AZURE_LOGTAG);
+    expect_string(__wrap__mterror, formatted_msg, "This is a Critical");
+
+    wm_parse_output(output_module, WM_AZURE_LOGGING_TOKEN, WM_AZURE_LOGTAG, NULL);
+
+}
+
+static void test_wm_parse_output_all_with_error(void **state) {
+    char * output_module = {
+        ":azure_wodlxxxx: - DEBUG - Setting 1 thread to pull 100 messages in total\n"
+        ":azure_wodlexxx: - INFO - Received and acknowledged 0 messages\n"
+        ":azure_wodlexxx: - ERROR - This is an Error\n"
+        ":azure_wodlexxx: - WARNING - This is a Warning\n"
+        ":azure_wodlexxx: - CRITICAL - This is a Critical\n"
+        };
+
+    will_return(__wrap_isDebug, 1);
+
+    wm_parse_output(output_module, WM_AZURE_LOGGING_TOKEN, WM_AZURE_LOGTAG, NULL);
+
+}
+
+static void test_msg_to_print_according_to_debugLevel(void **state) {
+    char * tokenized_line = {
+        "- DEBUG - Setting 1 thread to pull 100 messages in total"
+        };
+
+    char buff[65535] ={0};
+    char * service_title = "az-title";
+
+    if(msg_to_print_according_to_debugLevel(buff, tokenized_line, W_STR_DEBUG, NULL)) {
+        assert_string_equal(buff, "Setting 1 thread to pull 100 messages in total");
+    }
+
+    memset(buff, 0, 65535);
+    tokenized_line = "- DEBUG - Setting 1 thread to pull 100 messages in total";
+    if(msg_to_print_according_to_debugLevel(buff, tokenized_line, W_STR_DEBUG, service_title)) {
+        assert_string_equal(buff, "az-title Setting 1 thread to pull 100 messages in total");
+    }
+}
+
+static void test_msg_to_print_according_to_debugLevel_error(void **state) {
+    char * tokenized_line = {
+        "- DEBUGXXXX - Setting 1 thread to pull 100 messages in total"
+        };
+
+    char buff[65535] = {0};
+
+    if(0 == msg_to_print_according_to_debugLevel(buff, tokenized_line, W_STR_DEBUG, NULL)) {
+        assert_string_equal(buff, "");
+    }
+}
+
+int main(void) {
+    const struct CMUnitTest tests_parser_output[] = {
+        /*wm_parser_output  */
+        cmocka_unit_test(test_wm_parse_output_Dlevel0),
+        cmocka_unit_test(test_wm_parse_output_Dlevel1),
+        cmocka_unit_test(test_wm_parse_output_all_with_error),
+        cmocka_unit_test(test_msg_to_print_according_to_debugLevel),
+        cmocka_unit_test(test_msg_to_print_according_to_debugLevel_error),
+        cmocka_unit_test(test_wm_parse_output_with_error)
+    };
+
+    int result;
+    result = cmocka_run_group_tests(tests_parser_output, NULL, NULL);
+    return result;
+}
```

Comment:
Missing empty line at the end of the file.

Reply:
done! https://github.com/wazuh/wazuh/pull/19032/commits/8304b476339d6e1ef2efe2be045a79ee9d5f5357